### ISSUE TITLE
potential-bugs InvalidRange does not work in all cases

### ISSUE THREAD

**Issue by kepocnhh** [arturbosch/detekt#2044]
```kotlin
private fun verifyInvalidRange() {
    for (i in 2 until 1) {} // empty and detekt work
    for (i in 2 until 2) {} // empty but detekt does not work
    for (i in 2 until 3) {} // not empty
}
```
Documentation of `Int.until(to: Int): IntRange`:
> If the [to] value is less than or equal to `this` value, then the returned range is empty.
- `2 until 1` - 1 is less than or equal to 2
- `2 until 2` - 2 is less than or equal to 2

## Expected Behavior
The rule will work and the detekt will report an error in case of `2 until 2`.

## Observed Behavior
Detekt does not report an error in case of `2 until 2`.

## Steps to Reproduce
I use my task:
```kotlin
task<Detekt>("verifyQuality") {
    source = files(subprojects.withPlugin("kotlin").srcDirs("main")).asFileTree
    config = files(
        "buildSrc/src/main/resources/detekt/config/potential_bugs.yml"
    )
    reports {
        html {
            enabled = true
            destination = File(analysisQualityHtmlPath)
        }
        xml.enabled = false
        txt.enabled = false
    }
}
```

File potential_bugs.yml contains:
```yaml
build:
  maxIssues: 0

potential-bugs:
  active: true
  InvalidRange:
    active: true

processors:
  active: false
```

## Context
The problem is only with the `until` method and only for the case when both parameters are equivalent. For example, the `downTo` method is correctly determined by detekt. Apparently there is a problem in the code of the `io.gitlab.arturbosch.detekt.rules.bugs.InvalidRange` class.
```kotlin
private fun checkRangeTo(lower: Int, upper: Int) = lower > upper
private fun checkUntil(lower: Int, upper: Int) = lower > upper
```
The checks for both methods are the same. Although `downTo` creates an `IntRange` with the parameter `endInclusive`, and `until` with the parameter `endInclusive-1`.
In all other cases, I did not find any problems with this rule.

## Environment
java 1.8.0_221
kotlin 1.3.50
detekt 1.1.1
Gradle 5.5.1
macOS Mojave Version 10.14.6 (18G87)

**Comment by schalkms** [comment]
Thanks for analyzing this problem.
